build_tools/cmake/iree_tablegen_doc.cmake

```diff
@@ -70,9 +77,30 @@ function(iree_tablegen_doc)
     list(GET _RULE_OUTS 0 _OUTPUT)
     list(REMOVE_AT _RULE_OUTS 0)
 
-    # TableGen this output with the given command.
-    tablegen(${_TBLGEN} ${_OUTPUT} ${_COMMAND} ${_DIALECT} ${_INCLUDE_DIRS} ${_FLAGS})
-    list(APPEND _OUTPUTS ${CMAKE_CURRENT_BINARY_DIR}/${_OUTPUT})
+    set(_OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/${_OUTPUT})
+
+    # Create add_custom_command with depfile support for Ninja.
+    if(CMAKE_GENERATOR MATCHES "Ninja")
+      add_custom_command(
+        OUTPUT ${_OUTPUT_FILE}
+        COMMAND ${_TBLGEN_EXE} ${_COMMAND} ${_DIALECT} ${_INCLUDE_DIRS} ${_FLAGS} ${_TD_FILE_ABS}
+                --write-if-changed -o ${_OUTPUT_FILE} -d ${_OUTPUT_FILE}.d
+        DEPENDS ${_TBLGEN_TARGET} ${_TBLGEN_EXE} ${_TD_FILE_ABS}
+        DEPFILE ${_OUTPUT_FILE}.d
```

Comment:
Is DEPFILE ninja only? I'm a complete noob at this build file stuff and don't know why this would be ninja only (or if we care about other generators).